File: CentralMusic/CentralMusicAPI/Models/Users/UserRegister.cs
using CentralMusicAPI.Entities;
using System.ComponentModel.DataAnnotations;

namespace CentralMusicAPI.Models.Users
{
    /// <summary>
    /// Modelo usado por um utilizador
    /// para se resgistar com os seus dados pessoais
    /// </summary>
    public class UserRegister
    {
        [EmailAddress]
        [Required]
        public string Email { get; set; }
        [Required]
        public string Password { get; set; }
        [Required]
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        [Required]
        public Address Localization { get; set; }
    }
}

File: CentralMusic/CentralMusicFlutter/central_music/lib/models/user_register.dart
import 'package:central_music/models/address.dart';

class UserRegister {
  final String email;
  final String password;
  final String firstName;
  final String lastName;
  final Address localization;

  UserRegister({
    this.email,
    this.password,
    this.firstName,
    this.lastName,
    this.localization});

  factory UserRegister.fromJson(Map<String, dynamic> json) {
    return UserRegister(
      email: json["email"],
      password: json["password"],
      firstName: json["firstName"],
      lastName: json["lastName"],
      localization: Address.fromJson(json["localization"]));
  }
}